src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4851,8 +4853,26 @@ trait Types
   private[scala] val boundsContainType = (bounds: TypeBounds, tp: Type) => bounds containsType tp
   private[scala] val typeListIsEmpty = (ts: List[Type]) => ts.isEmpty
   private[scala] val typeIsSubTypeOfSerializable = (tp: Type) => tp <:< SerializableTpe
-  private[scala] val typeIsNothing = (tp: Type) => tp.typeSymbolDirect eq NothingClass
-  private[scala] val typeIsAny = (tp: Type) => tp.typeSymbolDirect eq AnyClass
+  private[scala] val typeIsNothing = (tp: Type) => {
```

Comment:
Boyscout opportunity: turn these into methods. It seems backwards to me that we're creating function values for use in `toString` (as argument for `filterNot`), while all other call sites are direct invocations which would be more efficient if this is defined as a method.

Reply:
Just these two? Or all of them?

Reply:
I haven't looked at the callsites for all of them, and I may be missing another reason these are here (undocumented optimization), so let's be conservative and only do these two. /cc @retronym on thoughts re: performance impact